libmiemu/controller.cpp

```diff
@@ -1 +1,379 @@
 #include "controller.h"
+
+void Controller::signals_in(uint16_t instruction)
+{
+	_instruction = instruction;
+	_opcode = _instruction & 0xF000;
```

Comment:
You need to `>>` right shift this as well or it won't fit in the 8-bit `_opcode`.